### ISSUE TITLE
Consider dropping support for unmaintained Node versions

### ISSUE THREAD

**Issue by demurgos** [dlmanning/gulp-sass#648]
Hi,
I am opening this issue following [this comment](https://github.com/dlmanning/gulp-sass/pull/647#discussion_r158902063).

Here is [Node's maintenance calendar](https://github.com/nodejs/Release). After _iojs_ was merged back into Node, they started a cyclic maintenance schedule. Today, only the versions `>= 4` are maintained by Node.

This library runs CI against Node 0.10, stable, and iojs (Node 3). Sass only runs CI for Node >= 4 but it seems that they are still testing it manually with Node 0.10.

When the older versions of Node reached end of life, there was a move in the ecosystem to adopt the new ES6 syntax. The most visible change is the use of `const` and `let` declarators. This means that a large (and still growing) part of npm is no longer usable with old Node versions.
Trying to still support these versions becomes difficult, it will always offer a second-class experience and may also prevent efforts to encourage users to stay up to date.

I would recommend a semver major update, announcing that Node 4+ is required for the gulp plugin and updating the Travis config to use:
```
node_js:
  - "lts/argon"
  - "lts/boron"
  - "lts/carbon"
  - "node"
```

Each year, an LTS version will now reach end of life. It would also be good to know what is the policy around it. I'd recommend to not break support for this version for the sake of it, but if an update is needed and requires newer versions, then there should be no big problem to just increase the semver major number and require maintained LTS versions.

**Comment by demurgos** [comment]
After some tests, it appears that **`node-sass` does not support Node < 4.** When downloading binaries, it uses `request` which uses `hoek`. This module requires Node 4 so `node-sass` itself requires Node 4.

**Comment by xzyfer** [comment]
Fixed in #667
